sql/moz-fx-data-shared-prod/org_mozilla_firefox/event_types_update_v1/query.sql

```diff
@@ -0,0 +1,15 @@
+DECLARE submission_date DATE DEFAULT @submission_date;
+
+EXECUTE IMMEDIATE CONCAT(
+  """CREATE OR REPLACE VIEW
+    org_mozilla_firefox.event_types
+  AS
+  SELECT
+    * EXCEPT (submission_date)
+  FROM
+    org_mozilla_firefox_derived.event_types_v1
+  WHERE
+    submission_date = '""",
+  CAST(submission_date AS STRING),
+  "'"
+);
```

Comment:
It would reduce complexity, but also add additional risk. This table is small, and this approach was kept so that we had a valid history we could walk through in case something went wrong. The query that updates this table is _very_ complicated so I wanted to ensure rollbacks and backfills were easy.

Reply:
Another way we could express this table is by having it partitioned by day, but each day includes only the new entries. So the incremental query would references `FROM event_types_v1 WHERE submission_date < @submission_date` and return only `new_event_types` rather than `all_event_types` to populate the day's partition. The view would then be a simple `SELECT *` without a filter on date.

That setup avoids needing to generate the view, but provides better characteristics for rewinding.

Reply:
Then I'd advocate for turning the existing `event_types_v1` into a partitioned `event_types_history_v1` and then `event_types_v1` be an unpartitioned table that is rewritten every day. This would avoid having this surprising `EXECUTE IMMEDIATE CONCAT` query.sql that doesn't actually populate a table.

Reply:
Hmm, then we'd end up with 3 queries: `event_types_history_v1`, `event_types_v1`, and `event_types` view. That adds complexity in a different way.

Reply:
We may be able to make this a query, rather than an `EXECUTE IMMEDIATE`. I'm trying to recall the reason we used that formation.

Reply:
I assume a view definition can't contain a query parameter, so the EXECUTE IMMEDIATE is necessary to inject the query param into the view as a literal date.

> Hmm, then we'd end up with 3 queries: event_types_history_v1, event_types_v1, and event_types view. That adds complexity in a different way.

Yes, it involves two table + a view rather than one table + a view, but in the current formulation this PR contains a pseudo-query that appears under a directory path that looks like it should produce a table called `event_types_update_v1` but in fact no such table will ever exists. I would argue that this is corrupting the intent of the directory structure in order to plug in to the DAG generation machinery.

At the very least, this file should be changed to be called `script.sql` rather than `query.sql` to give some hint that it's not actually running a query that populates a table.

But again, I would prefer to introduce another table in the chain if that gets us to a state that's more obviously comprehensible for future developers.

Reply:
Nope, that does not work: `Error in query string: Error processing job 'frank-sandbox:bqjob_r1c34be6c2896468c_00000175be402170_1': Query error: Query parameters cannot be used inside SQL view bodies at [11:21]`

Reply:
Okay, given the current options I do think it's wise to create `events_daily_v1` as a materialized view on the most recent day -- I did hate the `EXECUTE IMMEDIATE`. Let me update the PR.